Tools/src/NtupleFiller.cc

```diff
@@ -203,12 +319,161 @@ namespace jmt {
     return which_pv;
   }
 
-  void TracksSubNtupleFiller::operator()(const edm::Event& e, JetsSubNtupleFiller* jf, PrimaryVerticesSubNtupleFiller* vf, BeamspotSubNtupleFiller* bf) {
+  
+  bool TracksSubNtupleFiller::ismu(const edm::Event& e, MuonsSubNtupleFiller* mf, reco::TrackRef& tk) {
+    bool ismu = false;
+    
+    auto ms = mf->muons(e);
+    std::vector<reco::TrackRef> r;
+    for (size_t m = 0, me = ms.size(); m < me; ++m) {
+     
+      reco::TrackRef mtk = ms[m].track();
+      
+      if (!ms[m].track().isNull()) {
+
+	if (mtk->pt() > 1) {
+	  r.push_back(mtk);
+	}
+      }
+    }
+    
+    for (size_t j = 0, je = r.size(); j < je; ++j) {
+      double dr = reco::deltaR(tk->eta(), tk->phi(), r[j]->eta(), r[j]->phi());
+      if (dr < 0.001 ) {
+	ismu = true;
+      }
+    }
+    return ismu;
+  }
+
+
+  
+  bool TracksSubNtupleFiller::isel(const edm::Event& e, ElectronsSubNtupleFiller* ef, reco::TrackRef& tk) {
+    bool isel = false;
+
+    auto es = ef->electrons(e);
+   
+    std::vector<reco::GsfTrackRef> r;
+    for (size_t e = 0, ee = es.size(); e < ee; ++e) {
+      
+      reco::GsfTrackRef etk = es[e].gsfTrack(); 
+      
+      if (!es[e].gsfTrack().isNull()) {
+	if (etk->pt() > 1) {
+	  r.push_back(etk);
+	}
+      }
+    }
+      
+    for (size_t j = 0, je = r.size(); j < je; ++j) {
+      double dr = reco::deltaR(tk->eta(), tk->phi(), r[j]->eta(), r[j]->phi());
+      if (dr < 0.0001 ) {
+	isel = true;
+      }
+    }
+    
+    return isel;
+  }
+  
+  //good muon passes all lepton selection criteria EXCEPT for pt (just the standard > 1 GeV)
+  bool TracksSubNtupleFiller::isGoodMu(const edm::Event& e, MuonsSubNtupleFiller* mf, reco::TrackRef& tk) {
+    bool isgoodmu = false; 
+    auto ms = mf->muons(e);
+    std::vector<reco::TrackRef> r;
+    
+    for (size_t m = 0, me = ms.size(); m < me; ++m) {
+      
+      reco::TrackRef mtk = ms[m].track();
+
+      if (!ms[m].track().isNull()) {
+	const float iso = (ms[m].pfIsolationR04().sumChargedHadronPt + std::max(0., ms[m].pfIsolationR04().sumNeutralHadronEt + ms[m].pfIsolationR04().sumPhotonEt -0.5*ms[m].pfIsolationR04().sumPUPt))/ms[m].pt();
+
+	if (mtk->pt() > 1) {
+	  if (mtk->eta() < 2.4) {
+	    if (ms[m].passed(reco::Muon::CutBasedIdMedium)) {
+	      if (iso < 0.15) {
+		r.push_back(mtk);
+	      }
+	    }
+	  }
+	}
+      }
+    }
```

Comment:
@abbywarden and @pkotamnives, a naive question: we also apply lepton ID requirements in https://github.com/DisplacedVertices/cmssw-usercode/blob/UL_Lepton/Tools/python/PATTupleSelection_cfi.py. Are the leptons that make it here a subset of those from the PATTupleSelection, or does it just ignore that cfi file? Just want to be sure that we know exactly what we are doing, especially if e.g. Peace's WH/ZH studies use a different ID criteria than Abby's displaced lepton studies. This all also reminds me that we'll need to apply lepton efficiency correction scale factors eventually (since I don't think we are yet?), though it certainly won't be in this part of the code.